datadog_checks_base/datadog_checks/base/checks/win/wmi/__init__.py

```diff
@@ -2,13 +2,21 @@
 # All rights reserved
 # Licensed under a 3-clause BSD style license (see LICENSE)
 from collections import namedtuple
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
 
 from six import iteritems
 
+from datadog_checks.base import ConfigurationError
+from datadog_checks.base.errors import CheckException
+
 from ... import AgentCheck
 from .sampler import WMISampler
 
 WMIMetric = namedtuple('WMIMetric', ['name', 'value', 'tags'])
+WMIProperties = Tuple[Dict[str, Tuple[str, str]], Iterable[str]]
```

Comment:
Given that `properties` is defined as a `map`, and `map` can only be iterated through once (second time it yields nothing), this should probably be…

```suggestion
WMIProperties = Tuple[Dict[str, Tuple[str, str]], Iterator[str]]
```

We'd need to update other references to `Iterable` properties too.